crates/pep440-rs/src/version_specifier.rs

```diff
@@ -478,6 +495,16 @@ impl VersionSpecifier {
             (Bound::Included(v1), Bound::Included(v2)) if v1 == v2 => {
                 (Some(VersionSpecifier::equals_version(v1.clone())), None)
             }
+            // `v >= 3.7 && v < 3.8` is equivalent to `v == 3.7.*`
+            (Bound::Included(v1), Bound::Excluded(v2))
+                if v1.release().len() == 2
+                    && v2.release() == [v1.release()[0], v1.release()[1] + 1] =>
+            {
+                (
+                    Some(VersionSpecifier::equals_star_version(v1.clone())),
+                    None,
+                )
+            }
```

Comment:
I think this isn't true _in general_... I think it's only true if you ignore pre-releases and such.

Reply:
Renamed this to `VersionSpecifier::from_release_only_bounds`.